sources/window/error_source.go

```diff
@@ -96,124 +73,97 @@ type healthyIfNotAllErrorsSource struct {
 	}
 
 	return source, nil
 }
 
 // Submit submits an error.
-func (h *healthyIfNotAllErrorsSource) Submit(err error) {
-	h.sourceMutex.Lock()
-	defer h.sourceMutex.Unlock()
+func (e *errorHealthCheckSource) Submit(err error) {
+	e.sourceMutex.Lock()
+	defer e.sourceMutex.Unlock()
 
 	// If using anchored windows when last submit is greater than the window
 	// it will re-anchor the next window with a new repairing deadline.
-	if !h.hasSuccessInWindow() && !h.hasErrorInWindow() {
-		newRepairingDeadline := h.timeProvider.Now().Add(h.repairingGracePeriod)
-		if newRepairingDeadline.After(h.repairingDeadline) {
-			h.repairingDeadline = newRepairingDeadline
+	if !e.hasSuccessInWindow() && !e.hasErrorInWindow() {
+		newRepairingDeadline := e.timeProvider.Now().Add(e.repairingGracePeriod)
+		if newRepairingDeadline.After(e.repairingDeadline) {
+			e.repairingDeadline = newRepairingDeadline
 		}
 	}
 
 	if err != nil {
-		h.lastError = err
-		h.lastErrorTime = h.timeProvider.Now()
+		e.lastError = err
+		e.lastErrorTime = e.timeProvider.Now()
 	} else {
-		h.lastSuccessTime = h.timeProvider.Now()
+		e.lastSuccessTime = e.timeProvider.Now()
 	}
 }
 
 // HealthStatus polls the items inside the window and creates the HealthStatus.
-func (h *healthyIfNotAllErrorsSource) HealthStatus(ctx context.Context) health.HealthStatus {
-	h.sourceMutex.RLock()
-	defer h.sourceMutex.RUnlock()
+func (e *errorHealthCheckSource) HealthStatus(ctx context.Context) health.HealthStatus {
+	e.sourceMutex.RLock()
+	defer e.sourceMutex.RUnlock()
 
 	var healthCheckResult health.HealthCheckResult
-	if h.hasSuccessInWindow() {
-		healthCheckResult = sources.HealthyHealthCheckResult(h.checkType)
-	} else if h.hasErrorInWindow() {
-		if h.lastErrorTime.Before(h.repairingDeadline) {
-			healthCheckResult = sources.RepairingHealthCheckResult(h.checkType, h.lastError.Error(), sources.SafeParamsFromError(h.lastError))
-		} else {
-			healthCheckResult = sources.UnhealthyHealthCheckResult(h.checkType, h.lastError.Error(), sources.SafeParamsFromError(h.lastError))
-		}
+	if e.hasSuccessInWindow() && e.errorMode == HealthyIfNotAllErrors {
+		healthCheckResult = sources.HealthyHealthCheckResult(e.checkType)
+	} else if e.hasErrorInWindow() {
+		healthCheckResult = e.getFailureResult()
 	} else {
-		healthCheckResult = sources.HealthyHealthCheckResult(h.checkType)
+		healthCheckResult = sources.HealthyHealthCheckResult(e.checkType)
 	}
 
 	return health.HealthStatus{
 		Checks: map[health.CheckType]health.HealthCheckResult{
-			h.checkType: healthCheckResult,
+			e.checkType: healthCheckResult,
 		},
 	}
 }
 
-func (h *healthyIfNotAllErrorsSource) hasSuccessInWindow() bool {
-	return !h.lastSuccessTime.IsZero() && h.timeProvider.Now().Sub(h.lastSuccessTime) <= h.windowSize
+func (e *errorHealthCheckSource) getFailureResult() health.HealthCheckResult {
+	params := map[string]interface{}{
+		"error": e.lastError.Error(),
```

Comment:
This no longer includes all of the safe params from the last error as was done previously, which leads to the differing test assertions. Is this expected, and if so why leave them off?

Reply:
The error message already contains all of the params (werror adds them on `.Error()`). It doesn't add much value to add them as params as it's not a common workflow to aggregate on health check params.
Instead I wanted it to be consistent with the multi key error health checks were we do something similar (just make the params be the error messages).

Reply:
Cool that works with me